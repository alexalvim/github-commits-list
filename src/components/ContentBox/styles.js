import styled from 'styled-components';

import Colors from '../../colors';
import Spaces from '../../spaces';
import Typo from '../../typo';

const ContentWrapper = styled.div`
  background-color: ${Colors.lightestColor};
  border: solid 1px ${Colors.lightGray};
  border-radius: 5px;
  padding: ${Spaces.double}
  width: 100%;

  ${(props) => props.wrapperStyles}
`;

const Title = styled.h2`
  color: ${Colors.darkestColor};
  font-size: ${Typo.large};
`;

const TitleContainer = styled.div`
  align-items: center;
  display:flex;
  margin-bottom: ${Spaces.base};
`;

const RightContentContainer = styled.div`
  margin-left: auto;
`;

export {
  ContentWrapper,
  Title,
  TitleContainer,
  RightContentContainer
}