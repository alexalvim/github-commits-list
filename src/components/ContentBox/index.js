import React from 'react';

import { ContentWrapper, Title } from './styles';

export default ({ title, wrapperStyles, children }) => (
  <ContentWrapper wrapperStyles={wrapperStyles}>
    <Title>{title}</Title>
    {children}
  </ContentWrapper>
)