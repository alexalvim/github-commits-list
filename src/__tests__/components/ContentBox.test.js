import React from 'react';
import { shallow } from 'enzyme';

import ContentBox from '../../components/ContentBox';

describe('Testing ContentBox', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ContentBox title='test' />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with right content', () => {
    const wrapper = shallow(<ContentBox title='test' rightContent={<div id='test'>Test</div>} />);

    expect(wrapper.find('#test').length).toEqual(1);
  });

  it('should render correctly with children', () => {
    const wrapper = shallow(
      <ContentBox title='test'>
        <div id='children'>Children</div>
      </ContentBox>
    );

    expect(wrapper.find('#children').length).toEqual(1);
  })
});