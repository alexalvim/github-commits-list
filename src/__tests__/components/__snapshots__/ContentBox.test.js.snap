// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ContentBox should render correctly 1`] = `ShallowWrapper {}`;
